### DISCUSSION TITLE
not detecting birds but constantly analyzing

### DISCUSSION THREAD

**Discussion by Klages** [mcguirepr89/BirdNET-Pi#721] [Q&A] [answered]
For 3 days, my birdnetpi has not detected a single bird. I thought maybe a service failed, so I restarted the Pi but still no detections.
I checked the logs where I saw that the Pi constantly analyzes audio, even when I plug out the microphone. 
The application is completely updated (I'm not sure anymore when I last updated and if it could be coherent), also I didn't make any changes to the configuration. What could be the cause for this, and how can I fix the problem?

log with microphone unplugged:
![image](https://user-images.githubusercontent.com/3393617/216167147-a4b5f391-73ab-4e1e-a78f-5d804f457634.png)

the system seems to be fine otherwise:
![image](https://user-images.githubusercontent.com/3393617/216167371-7e752007-3e8d-4c87-b892-0eb31da42abf.png)

services are all running:
![image](https://user-images.githubusercontent.com/3393617/216167558-1a0322b9-0fc2-47c3-9953-692969af743d.png)
![image](https://user-images.githubusercontent.com/3393617/216167792-ed196b84-0776-4788-8690-8141e64c7d47.png)

**Comment by jds3d** [comment]
It looks to me like a microphone issue.  Listen to the audio it is recording (especially when a bird is present if you can find that).  If it sounds like a bird to you, I would be surprised.  Most likely there is a noise issue.

**Reply by Klages** [reply]
i've already listened to the microphone, at least when i talk into it, it sounds normal.

**Comment by Scraggarax** [comment, marked as answer]
Disable the microphone analysis service until it catches up.

That log is telling you that it's still Trying to catch up with the analysis.

Look at the log entry time (on the left) and the date/time (on the right) associated with the file that it is analyzing.

If it's constantly recording, then it won't keep up. I noted this when I experienced issue #653 and I have to keep stopping the recording service during the day or restarting the analysis service when it hangs (I've noticed keeping the logs open makes it faster but that might be anecdotal)

**Reply by jds3d** [reply]
Good catch.  it looks like it's speeding through old recordings now though--it's only taking him 1-2 seconds to process 15s of audio.

**Reply by ehpersonal38** [reply]
Good answer, and he won't necessarily need to stop the recording service ^ unless his disk is completely full. To be clear this is probably what happened:

1. BirdNET-Pi Analysis Service encounters _the notorious bug_ which causes it to hang/fail suddenly.
2. BirdNET-Pi Recording Service keeps producing audio files, but since the Analysis Service is down they never get processed and deleted.
3. Either 
**A)** You realize your Analysis Service isn't working anymore, so you restart all services which unhangs the service and fixes the issue. The Analysis Service will now start processing the huge backlog of audio files created by the Recording Service, which can take several hours or days. During this time your BirdNET-Pi will appear to be "broken," since you don't see any new detections, but it's not. (this is what's happening right now to Klages) 
or 
**B)** You don't realize in time that your Analysis Service isn't working anymore, so your Recording Service uses up all your disk space and just further compounds the issue

**Reply by Klages** [reply]
This was indeed the case. I disabled the microphone and waited for it to catch up, I took the whole night and I had to reboot several times as it would get stuck from time to time. Once there was even an exception. I could also see that the memory cache took a huge amount of my memory, but after everything ran through it finally detects birds again:

![image](https://user-images.githubusercontent.com/3393617/216251186-bd6182b5-9b87-4671-ba83-d0faf4a3c944.png)

Thanks everyone for the help!

**Reply by ProperTeaMogul** [reply]
I've just had this happen to me and the modular nature of the services meant it all continued to run and then caught up once I rebooted the services.

However, it got me wondering if there was a way for it to prompt or flag that maybe something wasn't working properly? That way you can leave it running in the background and just receive a notification e.g. email when it gets behind on processing by a certain amount.

**Comment by jds3d** [comment]
This looks like it's processing pre-dawn data now.  Perhaps once you get to dawn you will get real observations?

**Reply by Scraggarax** [reply]
Yeah, overnight it just kinda tries to guess what's there in the ambient sound, so there's usually nothing. I get the yellow rail overnight when it's raining sometimes lmao.

You'll get the observations displaying when the analysis service catches up to the time it was detected. Since the display is more a record rather than a live display.

**Comment by chrsaltz** [comment]
Came here to report basically the same issue with a third brand new installation. Started with a Pi 3B that was totally maxed out on CPU and RAM, and heat throttling. After days of sitting there, still "Your system is currently processing a backlog of audio. This can take several hours before normal functionality of your BirdNET-Pi resumes."

Just got a new 8GB Pi 4B in the mail today and loaded Birdnet on it. CPU and RAM have loads of headroom now, but still "Your system is currently processing a backlog of audio. This can take several hours before normal functionality of your BirdNET-Pi resumes."

Logs are similar to the OP where it keeps trying to detect non-existent birds (I'm inside at night in a quiet room) even with the mic unplugged

![image](https://user-images.githubusercontent.com/48190611/232271474-475e5750-4798-4b53-b251-a18ad83f79bb.png)

I've disabled the recording service, restarted all core services, rebooted the Pi, etc. Feels like a bug at this point.

**Reply by ehpersonal38** [reply]
So the "trying to detect non-existent birds" screenshot shows perfectly normal behavior, the analyzer runs 24/7, and you can see how low the confidence values for those birds are. The model can't return "no results," so it just returns a selection of birds that it has the highest confidence in, even if the confidence is very very low. 

As for the backlog issue, those log screenshots actually show your Pi isn't processing a backlog at all. Your BirdNET-Pi's processing/analysis looks perfectly normal there. So the issue must lie somewhere else. Something that would help me debug this is, when you see that "backlog" message on your Overview page, could you please go to "Today's Detections," and make sure the first 15 detections on the page have spectrogram images generated for them? Here's an example of a detection with spectrogram images:

![chrome_xF0XLOUq4q](https://user-images.githubusercontent.com/103586016/232322275-7e37b91e-f3e4-458f-83bd-fa9025102a76.png)

All 15 of the most recent detections should look like this. Is that the case on your system?

**Reply by chrsaltz** [reply]
Hey, thank you so much first of all for the fast reply.

Good to know that's normal behavior regarding the View Logs output.

So, there are no detections on "Today's Detections". I haven't had the
microphone connected in about 24 hours, and the device has never been
outside. Brand new Pi with brand new installation, although I also had the
same experience on my old Pi too (it was never outside since I didn't get
past the "testing" phase). I did play a couple of common bird songs at it
from my phone  last night but it didn't detect those anyway due to the backlog.

![image](https://user-images.githubusercontent.com/48190611/232366143-cd7dc280-a7af-4281-ac5a-af731fcbe829.png)

After sitting overnight and all day, still showing the same message on the
home page.

![image](https://user-images.githubusercontent.com/48190611/232366177-2f09ffff-1fa3-41d2-aef9-aa56fd038943.png)

**Reply by jaredb7** [reply]
Not sure if this is the issue but I noticed in your last screenshot that it looks like you don't have any GPS coordinates set (0.XX for both), I believe the detection system uses those to more accurately select birds that are common to your location or area.

They don't have the your exact coordinates but somewhere in your vicinity or city and that should be enough for it to work (if don't wish you input your real location)

Could you try setting them via Tools -> Settings, should be one of the first options.
Then restart core services, play bird call into the mic and see if it gets picked up?

**Reply by ehpersonal38** [reply]
@jaredb7 is right, without your latitude and longitude being set properly there's no chance your Pi will identify any birds at all. 

Also I think the backlog message is just a cosmetic issue at this point, I'll fix it soon. But there is no real backlog and your system is running well.

**Reply by chrsaltz** [reply]
Thank you both for the help again. Adding coordinates was definitely progress. I did that and updated to the newest version from the web interface, restarted the system. The "...currently processing..." message went away and the log is now doing phantom detections of species that exist in my area. 

Still no actual detections or spectrogram results when I play samples. I am going to plus my USB soundcard and microphone in to a laptop or something to make sure they actually work next.

**Reply by chrsaltz** [reply]
Ok, looking like a hardware issue at this point with my USB soundcard. The microphone vendor sent me a TRS cable and the soundcard only accepts TS input. Thank you again for the help!

**Reply by chrsaltz** [reply]
All working now after an adapter arrived in the mail. Thank you very much for your help @ehpersonal38 and @jaredb7
